helsingborg-stad/app-mitt-helsingborg
# Update the tab bar according to design

## Explain the changes you’ve made

Replace the current navbar with a new one following the [Figma design](https://www.figma.com/file/B0XpoR5OFMESSMRPzKGEoD/Mitt-HBG-%7C-Design-v3-%E2%80%A2-HT2020?node-id=23%3A2).
See task https://app.clickup.com/t/96ywny 

## Explain your solution

I add a new component, TabNavigator, which acts as a tab navigator following the api set by react-navigation. 
Then we can use this in the BottomBarNavigator instead of the default. The new TabNavigator in turn uses a small component called TabNavigatorItem to render each of the menu items. I get some issues between typescript types and proptypes, which I don't know how to fix.

I also add three new images, which are just grayscale versions of the original icons, to use when the link is inactive. Since these are png rather than svg icons, I think it has to be done like this. It would in general be better to have these icons as proper svg or some vector format. 

I also added some animation in the simplest way possible. If we want something fancier where the underline slides around, that can probably be done but it's a little harder. 

## How to test the changes?

Checkout this branch, and potentially run yarn ios again. Then the new navbar should show up in the app.

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots
new bottom navbar:
![Simulator Screen Shot - iPhone 11 - 2020-12-07 at 15 02 11](https://user-images.githubusercontent.com/7421890/101359953-4135d500-389d-11eb-8c68-d1ef1c6037fc.png)
